flink-java/src/main/java/org/apache/flink/api/java/io/CsvInputFormat.java

```diff
@@ -234,9 +228,21 @@ public OUT readRecord(OUT reuse, byte[] bytes, int offset, int numBytes) throws
 		}
 		
 		if (parseRecord(parsedValues, bytes, offset, numBytes)) {
-			// valid parse, map values into pact record
-			for (int i = 0; i < parsedValues.length; i++) {
-				reuse.setField(parsedValues[i], i);
+			if (pojoTypeClass == null) {
+				// result type is tuple
+				Tuple result = (Tuple) reuse;
+				for (int i = 0; i < parsedValues.length; i++) {
+					result.setField(parsedValues[i], i);
+				}
+			} else {
+				// result type is POJO
+				for (int i = 0; i < parsedValues.length; i++) {
+					try {
+						pojoFields[i].set(reuse, parsedValues[i]);
+					} catch (IllegalAccessException e) {
+						throw new RuntimeException(e);
```

Comment:
Add a message like `"Parsed value could not be set in POJO field \"" + pojoFieldsName[i] + "\""`